crates/rpc/rpc/src/eth/signer.rs

```diff
@@ -55,4 +85,130 @@ impl EthSigner for DevSigner {
     ) -> Result<TransactionSigned> {
         todo!()
     }
+
+    fn sign_typed_data(&self, address: Address, payload: &TypedData) -> Result<Signature> {
+        let encoded: H256 = payload.encode_eip712().map_err(|_| SignError::TypedData)?.into();
+        self.sign_hash(encoded, address)
+    }
+}
+#[cfg(test)]
+mod test {
+    use super::*;
+    use std::str::FromStr;
+    fn build_signer() -> DevSigner {
+        let addresses = vec![];
+        let secret =
+            SecretKey::from_str("4646464646464646464646464646464646464646464646464646464646464646")
+                .unwrap();
+        let accounts = HashMap::from([(Address::default(), secret)]);
+        let sign_engine = Secp256k1::new();
+        DevSigner { addresses, accounts, sign_engine }
+    }
+    #[tokio::test]
+    async fn test_sign_type_data() {
```

Comment:
should we replace these with tests from the EIP or from other clients?

Reply:
The data from this test was extracted from _geth_'s `signer/core/signed_data_test.go`. Should I add another one from EIP-712?